src/ui/components/Timeline/index.tsx

```diff
@@ -184,20 +209,26 @@ class Timeline extends Component<PropsFromRedux> {
 
     return (
       <div className="preview-markers-container">
-        {pointsForHoveredLineNumber.map((point: PointDescription, index: number) => (
-          <Marker
-            key={index}
-            point={point.point}
-            time={point.time}
-            hasFrames={!!point.frame}
-            location={point.frame?.[0]}
-            currentTime={currentTime}
-            hoveredPoint={hoveredPoint}
-            zoomRegion={zoomRegion}
-            overlayWidth={this.overlayWidth}
-            onSeek={() => {}}
-          />
-        ))}
+        {pointsForHoveredLineNumber.map((point: PointDescription, index: number) => {
+          const isPrimaryHighlighted = hoveredPoint?.point === point.point;
+          const isSecondaryHighlighted = getIsSecondaryHighlighted(hoveredPoint, point.frame?.[0]);
+
+          return (
+            <Marker
+              key={index}
+              point={point.point}
+              time={point.time}
+              hasFrames={!!point.frame}
+              location={point.frame?.[0]}
+              currentTime={currentTime}
+              isPrimaryHighlighted={isPrimaryHighlighted}
+              isSecondaryHighlighted={isSecondaryHighlighted}
+              zoomRegion={zoomRegion}
+              overlayWidth={this.overlayWidth}
+              onSeek={() => {}}
```

Comment:
this seems dangerous for re-renders. But i assume it is okay

Reply:
Dangerous how?

Reply:
its a new object everytime, so could trigger a re-render

Reply:
Ah, yes, we should definitely fix this before making `Marker` a `PureComponent`.